### PR TITLE
Get the iris/obsid module tests working

### LINKED ISSUE #8: Get the obsid module tests working
The code in the ```obsid``` module and its test module hasn't changed in the move from IRISpy.  But some of the infrastructure that was used to perform the tests may need editing due to the move, e.g. missing imports, changed path names, etc.

A PR addressing this issue should not try to get tests for other modules working as they will change substantially and the test can be fixed at those times.  However, a lot of the work done here may also mean there's less to do for other modules later in fixing their tests.

For simplicity, if this is the first issue addressed, this PR should comment out all other test modules by making those files giant docstrings.  The modules can be uncommented as work on their modules is performed.

Open PR to iris branch.

### PR DESCRIPTION
**Pull request by kakirastern**
Fixes #8. 

Tasks dealt with: 
1. Set up for `pytest`
2. Get the `iris/obsid` module tests working

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ resources:
   repositories:
   - repository: OpenAstronomy
     type: github
-    endpoint: ?????
+    endpoint: sunkit-instruments
```
Comment (nabobalis) [inline azure-pipelines.yml:13]:
```suggestion
    endpoint: sunpy
```

Comment (kakirastern) [reply azure-pipelines.yml]: Thanks @nabobalis!

**Comment by DanRyanIrish** [comment]
A number of the changes here are to do with the testing infrastructure.  I see @nabobalis has looked at this and left a comment which is great.  @nabobalis Did you see anything that could be a concern regarding the infrastructure here?

File: sunkit_instruments/iris/tests/test_spectrograph.py [hunk 2]
```diff
@@ -1,141 +1,142 @@
-
```
Comment (DanRyanIrish) [inline sunkit_instruments/iris/tests/test_spectrograph.py:1]: This file can be deleted for now.

File: tox.ini [hunk 3]
```diff
@@ -36,22 +33,61 @@ changedir = .tmp/{envname}
...
 deps =
+    # All our tests run in parallel which this plugin allows us to.
+    pytest-xdist
+    # The devdeps factor is intended to be used to install the latest developer version.
+    # of key dependencies.
+    devdeps: cython
+    devdeps: git+https://github.com/astropy/astropy
+    devdeps: git+https://github.com/sunpy/ndcube
+    devdeps: git+https://github.com/sunpy/sunpy
+    ndcube: git+https://github.com/sunpy/ndcube@1.3
```
Comment (DanRyanIrish) [inline tox.ini:56]: We don't need to test against the ndcube 1.3 branch in this package.  The stable version and master will suffice.

File: setup.cfg [hunk 4]
```diff
@@ -1,43 +1,78 @@
...
 author_email = sunpy@googlegroups.com
 license = BSD 3-Clause
 license_file = LICENSE.rst
 url = https://sunpy.org
 description = A sunpy-affiliated package for solar instrument-specific tools.
 long_description = file: README.rst
+long_description_content_type = text/x-rst
+platform = any
+classifiers =
+  Development Status :: 5 - Production/Stable
```
Comment (nabobalis) [inline setup.cfg:13]:
```suggestion
  Development Status :: 3 - Alpha
```

File: setup.cfg [hunk 5]
```diff
@@ -1,43 +1,78 @@
...
-install_requires = sunpy
-
-
+python_requires = >=3.7
+include_package_data = True
+setup_requires =
+    setuptools_scm
+install_requires =
+    sunpy
+    numpy
```
Comment (nabobalis) [inline setup.cfg:33]:
```suggestion
```

File: sunkit_instruments/iris/data/test/__init__.py [hunk 6]
```diff
@@ -6,11 +6,11 @@
 
 from astropy.utils.data import get_pkg_data_filename
 
-import irispy
+import sunraster
```
Comment (nabobalis) [inline sunkit_instruments/iris/data/test/__init__.py:9]: sunraster?

File: sunkit_instruments/iris/data/test/__init__.py [hunk 7]
```diff
@@ -6,11 +6,11 @@
...
 
 __all__ = ['rootdir', 'file_list', 'get_test_filepath']
 
-rootdir = os.path.join(os.path.dirname(irispy.__file__), "data", "test")
+rootdir = os.path.join(os.path.dirname(sunraster.__file__), "data", "tests")
```
Comment (nabobalis) [inline sunkit_instruments/iris/data/test/__init__.py:13]:
```suggestion
rootdir = os.path.join(os.path.dirname(sunkit_instruments.__file__), "data", "tests")
```

File: sunkit_instruments/iris/iris_tools.py [hunk 8]
```diff
@@ -4,26 +4,26 @@
...
 import scipy
 import scipy.io
-from scipy import ndimage
+# from scipy import ndimage
 from scipy import interpolate
 from sunpy.time import parse_time
 import sunpy.util.config
 from sunpy.util.net import check_download_file
-from ndcube import NDCube
+# from ndcube import NDCube
```
Comment (nabobalis) [inline sunkit_instruments/iris/iris_tools.py:26]: Why were these commented out?

File: sunkit_instruments/iris/tests/test_iris_tools.py [hunk 9]
```diff
@@ -1,308 +1,320 @@
-# -*- coding: utf-8 -*-
```
Comment (nabobalis) [inline sunkit_instruments/iris/tests/test_iris_tools.py:1]: What is gonna happen to this file?

File: sunkit_instruments/iris/tests/test_sji.py [hunk 10]
```diff
@@ -1,208 +1,209 @@
-# -*- coding: utf-8 -*-
```
Comment (nabobalis) [inline sunkit_instruments/iris/tests/test_sji.py:1]: What is gonna happen to this file?

File: tox.ini [hunk 11]
```diff
@@ -36,22 +33,61 @@ changedir = .tmp/{envname}
...
+     pytest --pyargs sunkit_instruments {toxinidir}/docs --cov sunkit_instruments --cov-config={toxinidir}/setup.cfg {posargs}
 
 [testenv:build_docs]
 changedir = docs
 description = invoke sphinx-build to build the HTML docs
 extras = docs
 commands =
-    sphinx-build -W -b html . _build/html {posargs}
-
+    # sphinx-build -W -b html . _build/html {posargs}
```
Comment (nabobalis) [inline tox.ini:88]:
```suggestion
```

File: tox.ini (hunk 3 above)
Comment (nabobalis) [inline tox.ini:56]:
```suggestion
```

File: tox.ini [hunk 12]
```diff
@@ -36,22 +33,61 @@ changedir = .tmp/{envname}
...
 
+# install testing framework
+# ... or install anything else you might need here
 deps =
+    # All our tests run in parallel which this plugin allows us to.
+    pytest-xdist
+    # The devdeps factor is intended to be used to install the latest developer version.
+    # of key dependencies.
+    devdeps: cython
+    devdeps: git+https://github.com/astropy/astropy
```
Comment (nabobalis) [inline tox.ini:53]:
```suggestion
```

File: tox.ini [hunk 13]
```diff
@@ -36,22 +33,61 @@ changedir = .tmp/{envname}
...
+    astropydev,build_docs,online: HOME = {envtmpdir}
 
+# install testing framework
+# ... or install anything else you might need here
 deps =
+    # All our tests run in parallel which this plugin allows us to.
+    pytest-xdist
+    # The devdeps factor is intended to be used to install the latest developer version.
+    # of key dependencies.
+    devdeps: cython
```
Comment (nabobalis) [inline tox.ini:52]:
```suggestion
```

File: tox.ini [hunk 14]
```diff
@@ -36,22 +33,61 @@ changedir = .tmp/{envname}
...
 description =
     run tests
+    devdeps: with the latest developer version of key dependencies
+    oldestdeps: with the oldest supported version of key dependencies
+setenv =
+    PYTHONPATH = sunkit-instruments/sunkit_instruments
+    MPLBACKEND = agg
+    COLUMNS = 180
+    PYTEST_COMMAND = pytest test.py
+    astropydev,build_docs,online: HOME = {envtmpdir}
```
Comment (nabobalis) [inline tox.ini:43]:
```suggestion
    build_docs: HOME = {envtmpdir}
```

File: tox.ini [hunk 15]
```diff
@@ -36,22 +33,61 @@ changedir = .tmp/{envname}
 #
 description =
     run tests
+    devdeps: with the latest developer version of key dependencies
+    oldestdeps: with the oldest supported version of key dependencies
+setenv =
+    PYTHONPATH = sunkit-instruments/sunkit_instruments
```
Comment (nabobalis) [inline tox.ini:39]:
```suggestion
```

File: tox.ini [hunk 16]
```diff
@@ -36,22 +33,61 @@ changedir = .tmp/{envname}
...
-commands =
-    pytest --pyargs sunkit_instruments {toxinidir}/docs --cov sunkit_instruments --cov-config={toxinidir}/setup.cfg {posargs}
+    tests
+# commands =
+#     !online: {env:PYTEST_COMMAND} {posargs}
+#     online: {env:PYTEST_COMMAND} --reruns 2 --timeout=180 --remote-data=any {posargs}
 
+# run the tests
+# ... or run any other command line tool you need to run here
+# May need to be modified to the following
```
Comment (nabobalis) [inline tox.ini:79]:
```suggestion
```

File: tox.ini [hunk 17]
```diff
@@ -36,22 +33,61 @@ changedir = .tmp/{envname}
...
 # The following indicates which extras_require from setup.cfg will be installed
+# dev is special in that it installs everything but we use tests here.
 extras =
-    test
-commands =
-    pytest --pyargs sunkit_instruments {toxinidir}/docs --cov sunkit_instruments --cov-config={toxinidir}/setup.cfg {posargs}
+    tests
+# commands =
+#     !online: {env:PYTEST_COMMAND} {posargs}
+#     online: {env:PYTEST_COMMAND} --reruns 2 --timeout=180 --remote-data=any {posargs}
```
Comment (nabobalis) [inline tox.ini:75]:
```suggestion
# commands =
```

File: tox.ini [hunk 18]
```diff
@@ -36,22 +33,61 @@ changedir = .tmp/{envname}
...
+    oldestdeps: with the oldest supported version of key dependencies
+setenv =
+    PYTHONPATH = sunkit-instruments/sunkit_instruments
+    MPLBACKEND = agg
+    COLUMNS = 180
+    PYTEST_COMMAND = pytest test.py
+    astropydev,build_docs,online: HOME = {envtmpdir}
 
+# install testing framework
+# ... or install anything else you might need here
```
Comment (nabobalis) [inline tox.ini:46]:
```suggestion
```

File: tox.ini [hunk 19]
```diff
@@ -36,22 +33,61 @@ changedir = .tmp/{envname}
 #
 description =
     run tests
+    devdeps: with the latest developer version of key dependencies
+    oldestdeps: with the oldest supported version of key dependencies
+setenv =
+    PYTHONPATH = sunkit-instruments/sunkit_instruments
+    MPLBACKEND = agg
+    COLUMNS = 180
+    PYTEST_COMMAND = pytest test.py
```
Comment (nabobalis) [inline tox.ini:42]:
```suggestion
    PYTEST_COMMAND = pytest --pyargs sunkit_instruments {toxinidir}/docs --cov sunkit_instruments --cov-config={toxinidir}/setup.cfg
```

File: tox.ini [hunk 20]
```diff
@@ -36,22 +33,61 @@ changedir = .tmp/{envname}
...
+    tests
+# commands =
+#     !online: {env:PYTEST_COMMAND} {posargs}
+#     online: {env:PYTEST_COMMAND} --reruns 2 --timeout=180 --remote-data=any {posargs}
 
+# run the tests
+# ... or run any other command line tool you need to run here
+# May need to be modified to the following
+commands =
+     pytest --pyargs sunkit_instruments {toxinidir}/docs --cov sunkit_instruments --cov-config={toxinidir}/setup.cfg {posargs}
```
Comment (nabobalis) [inline tox.ini:81]:
```suggestion
     {env:PYTEST_COMMAND} {posargs}
```

File: azure-pipelines.yml [hunk 21]
```diff
@@ -10,7 +10,7 @@ resources:
   repositories:
   - repository: OpenAstronomy
     type: github
-    endpoint: ?????
+    endpoint: sunpy
     name: OpenAstronomy/azure-pipelines-templates
     ref: master
```
Comment (nabobalis) [inline azure-pipelines.yml:16]: The test runs below need to be updated as well.

File: tox.ini [hunk 22]
```diff
@@ -1,6 +1,6 @@
 [tox]
 envlist =
-    py{36,37,38}
+    py{37,38}
     build_docs
     codestyle
 isolated_build = true
```
Comment (nabobalis) [inline tox.ini:6]: Remove the two lines below this one.

File: sunkit_instruments/iris/iris_tools.py (hunk 8 above)
Comment (kakirastern) [reply sunkit_instruments/iris/iris_tools.py:26]: These are not in use at the moment. I will uncomment them.

File: sunkit_instruments/iris/tests/test_iris_tools.py (hunk 9 above)
Comment (kakirastern) [reply sunkit_instruments/iris/tests/test_iris_tools.py:1]: @nabobalis Dan instructed me to comment this file out as a giant docstring for now

File: sunkit_instruments/iris/tests/test_sji.py (hunk 10 above)
Comment (kakirastern) [reply sunkit_instruments/iris/tests/test_sji.py:1]: @nabobalis Dan instructed me to comment this file out as a giant docstring for now

Comment (nabobalis) [reply sunkit_instruments/iris/tests/test_sji.py:1]: Sure

**Comment by kakirastern** [comment]
Will resume working on this starting next Monday

**Comment by nabobalis** [comment]
I believe most of this code was moved into the new IRISPy and updated. I am not sure it is worth updating the code in this repo.

**Comment by kakirastern** [comment]
Sure, not a problem...

**Comment by wtbarnes** [comment]
In light of the fact that irispy exists (is it [this](https://gitlab.com/LMSAL_HUB/iris_hub/irispy-lmsal) or [this](https://gitlab.com/LMSAL_HUB/iris_hub/iris_lmsalpy)?), should we remove any duplicate IRIS-related functionality here?

**Comment by nabobalis** [comment]
I would be in favour of that.

**Pull request closed without merging**